sdk/identity/azure-identity/src/azure_cli_credential.cpp

```diff
@@ -154,7 +154,10 @@ AccessToken AzureCliCredential::GetToken(
       try
       {
         return TokenCredentialImpl::ParseToken(
-            azCliResult, "accessToken", "expiresIn", "expiresOn");
+            azCliResult,
+            "accessToken",
+            "expiresIn",
+            std::vector<std::string>{"expires_on", "expiresOn"});
```

Comment:
The order of these two strings matter, correct?

We may be generalizing the approach prematurely by using a vector of strings, and it's introducing complexity.

I don't expect arbitrary set of strings to be passed in here, so we could add an extra parameter instead and simplify the parsing logic (no loops iterating through vectors and order/length checks).

Reply:
The order of parameters matters - the vector is parsed in order, the items are defined.
Vector is readable, and we don't need to handle the cases if the first parameter is empty, but the second one is set. Even if we had two parameters, the logic for applying them is exactly the same, so I would create an array inside the function and iterate over it.

I think vector parameter is better.